nzshm_common/util/compression.py

```diff
@@ -0,0 +1,43 @@
+"""Useful utility & helpers."""
+import base64
+import io
+import zipfile
+from pathlib import Path, PurePath
+from typing import Union
+
+
+def compress_path(source_path: str, output_zip: Union[str, PurePath]) -> str:
+    """Zip contents of source path and return the full archive path.
+
+    Handles both single file and a folder.
+    """
+    with zipfile.ZipFile(output_zip, "w", compression=zipfile.ZIP_DEFLATED) as zip:
```

Comment:
might be useful to have the compression method as an optional argument

Reply:
@voj Indeed - that can be an enhancement.